src/databudgie/utils.py

```diff
@@ -81,14 +81,25 @@ def join_paths(*components: Optional[str]) -> str:
     if len(real_components) == 1:
         return real_components[0]
 
-    first_component, *rest_components = real_components
+    bucket = None
     normalized_components = []
-    for c in rest_components:
+    for index, c in enumerate(real_components):
+        normalized_c = c
         if is_s3_path(c):
```

Comment:
why do we need to check each component to see if it's an S3 path?

Reply:
maybe I'm not understanding what the components are here

Reply:
this is any composition of s3 path components, i.e. root_location + location. so what do you expect to happen if root_location is `s3://bucket` and location were `s3://bucket/{table}`. because the answer would have previously duplicated the bucket into the path.

perhaps it should be an error, and could be, if we preferred that. but the current behavior is just confusing